### ISSUE TITLE
.nodecl is not implied with .importC on variables with the JS backend

### ISSUE THREAD

**Issue by PhilipWitte** [nim-lang/Nim#4233]
``` nim
type
  Thing = ref object
    value: int

var
  thing = Thing(value:123)

proc foo =
  # we need the .noDecl pragma here otherwise this line
  # generates `var this = null;` which is invalid JS code
  var this {.importC noDecl.}: Thing
  log $this.value

# need to call 'foo' here so it's generated in the JS output
foo() # prints: 'undefined'

# need to use `[0]` here cause global vars are stored in Arrays..
asm "`foo`.call(`thing`[0]);" # prints: '123'
```

The code comments explain the problem. The `{.noDecl.}` should be implied with `{.importC.}` like it is for `proc`, etc..

**Comment by yglukhov** [comment]
Fixed by #5297?

**Comment by Araq** [comment]
Indeed.
